削除機能の実装

ZYO-BAIKER opened the pull request:
what 商品詳細画面の「削除ボタン」をトップページへ遷移。
　　itemコントローラーにdestroyアクションを定義
why 出品した商品を取り消し可能にする為

ZYO-BAIKER committed:
削除機能の実装

ZYO-BAIKER commented:
編集機能を実装しようとしていたら、先に削除機能がうっかりできてしまったので、一度itemコントローラーのeditアクションをコメントアウトしている。

masahirokurauchi commented:
"こちらのプルリクエストですが、実際の画面や機能ができているか確認させていただきたいので、キャプチャ動画をご提出いただけますか？
「Gyazo　GIF　使い方」とgoogleで検索し、ツールによって生成されたURLをこのPRに貼っていただければ幸いです。
その後、再度レビュー依頼フォームにてご提出をお願いします。
お手数かけますが、よろしくお願いします。"

ZYO-BAIKER commented:
> "こちらのプルリクエストですが、実際の画面や機能ができているか確認させていただきたいので、キャプチャ動画をご提出いただけますか？
> 「Gyazo　GIF　使い方」とgoogleで検索し、ツールによって生成されたURLをこのPRに貼っていただければ幸いです。
> その後、再度レビュー依頼フォームにてご提出をお願いします。
> お手数かけますが、よろしくお願いします。"

gyazoも載せた筈ですが

yupimaru commented on app/controllers/items_controller.rb:
```diff
@@ -23,6 +23,16 @@ def show
+  # end
+
+  def destroy
+    item = Item.find(params[:id])
```
```ruby
item = Item.find(params[:id])
```

こちらの記述ですが、同様の記述を複数箇所で使用されているようです。
そのため、以下のようなメソッドを`private`以下に定義し、
`before_action`で呼び出せるように実装してみましょう。
また、その際`destroy`アクションで使用する変数名を`item`から`@item`へと変更しておきましょう。

```ruby
def (任意のメソッド名)
  @item = Item.find(params[:id])
end
```

yupimaru commented on app/controllers/items_controller.rb:
```diff
@@ -23,6 +23,16 @@ def show
+
+  def destroy
+    item = Item.find(params[:id])
+    item.destroy
+    redirect_to root_path
```
こちら、viewファイル同様、**出品したユーザーでないと削除できない**という条件分岐を記述しておきましょう。
万が一、出品者でない場合には商品詳細画面へ戻る等としておくと良さそうです。

yupimaru reviewed (commented):
レビュー依頼ありがとうございます。

Gyazoのご提出につきまして、大変失礼いたしました。
レビュー依頼フォームへ載せて頂いていましたね。確認させて頂きました。


コメントしましたので、ご確認をお願いいたします！

ZYO-BAIKER committed:
修正

before_actionでitem = Item.find(params[:id])をまとめた。
destroyアクションで、出品したユーザーでないと削除できないという条件分岐

chirikuzu commented on app/controllers/items_controller.rb:
```diff
@@ -1,5 +1,6 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!, only: [:new]
```
```ruby
before_action :authenticate_user!, only: [:new]
```
こちらdestroyおよびcreateアクションにも適用させてあげましょう。

chirikuzu reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします。

ZYO-BAIKER committed:
修正２

before_action :authenticate_user!にdestroyおよびcreateアクションを追加

haruka0410 commented on app/views/items/show.html.erb:
```diff
@@ -27,7 +27,7 @@
     <%= link_to '商品の編集', "#", method: :get, class: "item-red-btn" %>
     <p class='or-text'>or</p>
-    <%= link_to '削除', "#", method: :delete, class:'item-destroy' %>
+    <%= link_to '削除', item_path, method: :delete, class:'item-destroy' %>
```
```
<%= link_to '削除', item_path, method: :delete, class:'item-destroy' %>
```

item_pathですが、どの商品を削除するのかを特定するために引数で@itemを渡してあげる必要があるのではないかと考えられます。
ただ、送っていただいたGyazoだと正常に動いていますので、現状の記述でご確認いただき問題なければLGTMです。次の実装に進んでいただければと思います。

もし、不具合があれば修正後改めてレビュー依頼をお願いいたします。

haruka0410 reviewed (commented):
レビュー依頼ありがとうございます。
1点コメントいたしましたので、ご確認いただければと思います。
